src/module.c

```diff
@@ -10804,12 +10806,9 @@ void processModuleLoadingProgressEvent(int is_aof) {
     }
 }
 
-/* When a key is deleted (in dbAsyncDelete/dbSyncDelete/dbOverwrite), it 
+/* When a module key is deleted (in dbAsyncDelete/dbSyncDelete/dbOverwrite), it 
 *  will be called to tell the module which key is about to be released. */
 void moduleNotifyKeyUnlink(robj *key, robj *val, int dbid) {
-    server.lazy_expire_disabled++;
-    moduleNotifyKeyspaceEvent(NOTIFY_REMOVED,"removed",key,dbid);
-    server.lazy_expire_disabled--;
```

Comment:
i think i was expecting this code block to remain here, and just send a module event instead of a keyspace notification.
i.e. we still wanna catch all the execution flows we did before, and rely on our dictTwoPhaseUnlinkFind.
it's just that we decided this signal is an event rather than keyspace-notification.